### PR TITLE
NP-47439 FetchInstitutionReportHandler

### PR DESCRIPTION
**Pull request by monaullah**
Jira task: NP-47439  Generate nvi institution report based on data in nvi index

Part one, new handler `FetchInstitutionReportHandler`:
- Supports two media types: `application/vnd.ms-excel` and `application/vnd.openxmlformats-officedocument.spreadsheetml.sheet`
- Returns 401 Unauthorized if user does not have access right `MANAGE_NVI_CANDIDATES`
- Returns empty excel file (keeping it simple 😄 )

Next:
- Actually fill excel file with expected content

### CODE REVIEW

File: docs/openapi.yaml [hunk 1]
```diff
@@ -672,6 +672,52 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/Problem"
+  /institution-approval-report/{year}:
```
Comment (monaullah) [inline docs/openapi.yaml:675]: @ketilaa and I originally discussed using path `/institution-report/{year}` for `FetchInstitutionReportHandler`, but this path is already used for retrieving organization aggregations (`FetchInstitutionStatusAggregationHandler`).
Therefore named path `/institution-approval-report/{year}`

File: index-handlers/src/test/java/no/sikt/nva/nvi/index/apigateway/FetchInstitutionReportHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        var response = GatewayResponse.fromOutputStream(output, String.class);
+        assertThat(response.getHeaders().get(CONTENT_TYPE), is(mediaType));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+        "application/vnd.ms-excel",
+        "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+    })
+    void shouldReturnBase64EncodedOutputStreamWhenContentTypeIsExcel(String mediaType) throws IOException {
```
Comment (monaullah) [inline index-handlers/src/test/java/no/sikt/nva/nvi/index/apigateway/FetchInstitutionReportHandlerTest.java:97]: Docs: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-payload-encodings.html#api-gateway-payload-encodings-proxy

File: index-handlers/src/main/java/no/sikt/nva/nvi/index/xlsx/Excel.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Base64;
+import java.util.Base64.Encoder;
+import java.util.List;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+public record Excel(Workbook workbook) {
```
Comment (sondrev) [inline index-handlers/src/main/java/no/sikt/nva/nvi/index/xlsx/Excel.java:13]: Could use a better name.

**Review by sondrev — APPROVED** [review]

File: index-handlers/src/main/java/no/sikt/nva/nvi/index/xlsx/Excel.java (hunk 3 above)
Comment (sondrev) [reply index-handlers/src/main/java/no/sikt/nva/nvi/index/xlsx/Excel.java:13]: Something that describes what the record stores, or does.

**Review by torbjokv — APPROVED** [review]

File: index-handlers/src/main/java/no/sikt/nva/nvi/index/xlsx/Excel.java (hunk 3 above)
Comment (monaullah) [reply index-handlers/src/main/java/no/sikt/nva/nvi/index/xlsx/Excel.java:13]: @sondrev Any suggestions? `WorkbookWrapper`?

**Review by torbjokv — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**